src/lib/trpc/router.ts

```diff
@@ -64,537 +65,548 @@ import { prismaClient } from '$lib/server/prisma/client';
-	getPluginIdentifiers: t.procedure.query(async () => {
-		return getAllNeovimPluginNames();
-	}),
-	syncExistingNeovimConfig: t.procedure
-		.use(middlewares.isAuthenticated)
-		.input((input: unknown) => {
-			return z
-				.object({
-					owner: z.string(),
-					slug: z.string()
-				})
-				.parse(input);
-		})
-		.query(async ({ ctx, input }) => {
-			const user = ctx.getAuthenticatedUser();
-			const configBeforeSync = await getConfigBySlug(input.owner, input.slug);
-			if (!hasBeenOneDay(configBeforeSync.lastSyncedAt.toString())) {
-				throw new TRPCError({ code: 'FORBIDDEN' });
-			}
-			const token = await getGithubToken(user.id);
-			const config = await syncExistingRepoInfo(token, configBeforeSync);
-			await syncReadme(token, config);
-			const syncer = await getNeovimConfigSyncer(user, config);
-			return await syncer.treeSync();
-		}),
-	createNeovimConfig: t.procedure
-		.use(middlewares.isAuthenticated)
-		.input((input: unknown) => {
-			return z
-				.object({
-					initFile: z.nativeEnum(InitFileNames),
-					root: z.string(),
-					repo: z.string(),
-					branch: z.string()
-				})
-				.parse(input);
-		})
-		.mutation(async ({ ctx, input }) => {
-			const user = ctx.getAuthenticatedUser();
-			const token = await getGithubToken(user.id);
-			const tree = await fetchRepoFileTree(token, user.username, input.repo, input.branch);
-			validateConfigPath(tree, input.root ? `${input.root}/${input.initFile}` : input.initFile);
-			const config = await syncInitialRepoInfo(
-				user,
-				user.username,
-				input.repo,
-				input.root,
-				input.initFile
-			);
-			await syncReadme(token, config);
-			const syncer = await getNeovimConfigSyncer(user, config);
-			return await syncer.treeSync();
-		}),
-	findRepoInitFiles: t.procedure
-		.use(middlewares.isAuthenticated)
-		.input((input: unknown) => {
-			return z.object({ repo: z.string(), branch: z.string() }).parse(input);
-		})
-		.query(async ({ ctx, input }) => {
-			const user = ctx.getAuthenticatedUser();
-			const root = await getRepoFileTree(user, input.repo, input.branch);
-			const initFiles = new InitFileFinder().findAllInitFile(root);
-			return initFiles;
-		}),
+  syncPlugin: t.procedure
+    .use(middlewares.isAuthenticated)
+    .input((input: unknown) => {
+      return z
+        .object({
+          owner: z.string(),
+          name: z.string()
+        })
+        .parse(input);
+    })
+    .mutation(async ({ input: { owner, name }, ctx }) => {
+      const user = ctx.getAuthenticatedUser();
+      const syncer = await getPluginSyncer(user, owner, name);
+      return syncer.sync();
+    }),
+  toggleConfigMarkForDeletion: t.procedure
+    .use(middlewares.isAuthenticated)
+    .input((input: unknown) => {
+      return z.object({
+        id: z.number()
+      }).parse(input);
+    })
+    .mutation(async ({ input: { id }, ctx }) => {
+      const user = ctx.getAuthenticatedUser();
+      return await toggleConfigMarkedForDeletion(id, user.username);
```

Comment:
Instead of marking a config for deletion and subsequently removing it in the next sync. Could we instead immediately remove it here and remove the complexity regarding the sync logic?